File: cs214/projects/04/factorial.cpp
/* factorial.cpp computes and displays the
 * factorial of a number entered by the user.
 *
 * Input: an integer, n
 * Output: the value of n!
 * Begun by: Charles Blum, for cs214 at Calvin College
 * Begin on: 4 March 2014
 */

#include <iostream>
#include <cassert>

using namespace std;

double factorial(unsigned n) {
  double answer = 1.0;

  for (unsigned i = 2; i <= n; i++)
  {
    answer *= i;
  }
  return answer;
}

int main() {
  cout << "\nTo compute n!, enter n: ";
  unsigned n;
  cin >> n;

  assert(n >= 0);

  cout << n << "! = " << factorial(n) << endl;
}

File: cs214/projects/04/factorial.rb
#! /usr/bin/ruby
# factorial.rb displays the factorial of a number
#
# Begun by: Charles Blum, for CS 214 at Calvin College.
# Begun on: 4 March 2014
#
# Input:  n, an integer
# Precondition: n is >= 0
# Output: The factorial of n 

def comp_factorial(n)
  f = 1
  for i in 1..n
    f *= i
  end
  f
end

if __FILE__ == $0
  print "This program calculates the factorial of a number:"
  print "Enter a number n:"
  n = gets.chomp.to_i

  puts "#{n}! = #{comp_factorial(n)}"
end

